app/models/lookup_keys/variable_lookup_key.rb

```diff
@@ -2,7 +2,7 @@ class VariableLookupKey < LookupKey
   belongs_to :puppetclass, :inverse_of => :lookup_keys, :counter_cache => :variable_lookup_keys_count
 
   validates :puppetclass, :presence => true
-  validates :key, :uniqueness => true
+  validates :key, :uniqueness => true, :no_whitespace => true
```

Comment:
You can't edit the key for class parameters so I don't think it's needed there but I don't mind moving it.